code/modules/mob/living/carbon/human/npc/zizombies.dm

```diff
@@ -35,6 +35,7 @@
 	ADD_TRAIT(src, TRAIT_NOMOOD, TRAIT_GENERIC)
 	ADD_TRAIT(src, TRAIT_NOHUNGER, TRAIT_GENERIC)
 	ADD_TRAIT(src, TRAIT_NOSTAMINA, TRAIT_GENERIC)
+	ADD_TRAIT(src, TRAIT_STUNIMMUNE, TRAIT_GENERIC)//the undead shouldn't freeze in pain
```

Comment:
```suggestion
	ADD_TRAIT(src, TRAIT_NOPAINSTUN, TRAIT_GENERIC) //the undead shouldn't freeze in pain
```
While I do wholeheartedly agree that they shouldn't be stunned in pain, they should get a NOPAIN flag rather than NOSTUN, as Churn Undead, a spell specifically meant to deal with vampires and zizombies, stuns by design, and this would render them immune.